pandas/core/groupby.py

```diff
@@ -2536,8 +2582,12 @@ def ngroups(self):
 
     @cache_readonly
     def indices(self):
-        values = _ensure_categorical(self.grouper)
-        return values._reverse_indexer()
+        # for the situation of groupby list of groupers
```

Comment:
need to move this to the base class and inherit on the subclass

Reply:
I'm sorry but Grouping class inherits from object class. I don't really understand which method should be moved to base class

Reply:
you need to call a (private) method on the grouper here, which will then dispatch if its a BinGrouper or the BaseGrouper as appropriate. same as below.

Reply:
yes  you are misuderstanding

```
def indices(self):
     return self.grouper.get_indexer(self.grouper) # maybe better name
```

```
class BaserGrouper:
      def get_indexer(self, grouper):
             values = _ensure_categorical(self, grouper)
              return values._reverse_indexer()


class BinGrouper:
       def get_indexer(self, grouper):
           # grouper is self here
           labels = self.label_info
           uniques = self.result_index
           ...
```

this is just multiple dispatch

Reply:
But here comes another problem, grouper is not guaranteed to be a BaseGrouper or BinGrouper instance, it may be just a label ndarray, which I think is better not to add such a method.

Reply:
ok, fine for now. I think we should fix this though. We are a bit inconsistent where a Grouping object reassigns .grouper to a Base/Bin Grouper if needed. So Maybe should have a class which we can wrap up and hide details like these.

Reply:
I think the better way is to convert all groupby call to the format of [pd.Grouper, pd.TimeGrouper] so that the .grouper of Grouping instance is guaranteed to be Base/Bin Grouper instance, and easy to handle. But currently there are variety of ways to do the groupby scenario, I think it may cost time to do this refactor.

Reply:
sure, you can create an issue about this, or send a PR when convenient (after this one!)